src/handlers/chain_agnostic/route.rs

```diff
@@ -196,6 +196,15 @@ impl From<PrepareResponseAvailable> for PrepareResponseAvailableV1 {
     }
 }
 
+fn no_bridging_needed_response(initial_transaction: Transaction) -> Json<PrepareResponse> {
+    Json(PrepareResponse::Success(
+        PrepareResponseSuccess::NotRequired(PrepareResponseNotRequired {
+            initial_transaction: initial_transaction.clone(),
```

Comment:
Unnecessary `.clone()` I think

```suggestion
            initial_transaction,
```